soh/src/code/z_en_item00.c

```diff
@@ -1208,38 +1208,56 @@ void EnItem00_Draw(Actor* thisx, GlobalContext* globalCtx) {
     }
 }
 
-void EnItem00_CustomItemsParticles(Actor* Parent, GlobalContext* globalCtx, s16 getItemId) {
+void EnItem00_CustomItemsParticles(Actor* Parent, GlobalContext* globalCtx, GetItemEntry giEntry) {
     s16 color_slot;
-    switch (getItemId) {
-        case GI_MINUET_OF_FOREST:
-        case GI_SINGLE_MAGIC:
-        case GI_DOUBLE_MAGIC:
-            color_slot = 0;
-            break;
-        case GI_BOLERO_OF_FIRE:
-        case GI_DOUBLE_DEFENSE:
-            color_slot = 1;
-            break;
-        case GI_SERENADE_OF_WATER:
-            color_slot = 2;
-            break;
-        case GI_REQUIEM_OF_SPIRIT:
-            color_slot = 3;
-            break;
-        case GI_NOCTURNE_OF_SHADOW:
-            color_slot = 4;
-            break;
-        case GI_PRELUDE_OF_LIGHT:
-            color_slot = 5;
-            break;
-        case GI_STICK_UPGRADE_20:
-        case GI_STICK_UPGRADE_30:
-            color_slot = 6;
+    switch (giEntry.modIndex) {
+        case MOD_NONE:
+            switch (giEntry.itemId) {
+                case ITEM_SONG_MINUET:
+                    color_slot = 0;
```

Comment:
unrelated to this PR (mostly a note to self to make an issue for this) but it'd be good to not use magic numbers for the colors here

Reply:
made an issue https://github.com/HarbourMasters/Shipwright/issues/1156